editor/editor_properties_array_dict.cpp

```diff
@@ -776,9 +800,10 @@ void EditorPropertyDictionary::_change_type(Object *p_button, int p_index) {
 }
 
 void EditorPropertyDictionary::_add_key_value() {
-	// Do not allow nil as valid key. I experienced errors with this
-	if (object->get_new_item_key().get_type() == Variant::NIL) {
-		return;
+	// Treat empty object key as nil to prevent editor-only duplicates.
+	if (object->get_new_item_key().get_type() == Variant::OBJECT &&
+			object->get_new_item_key().is_null()) {
+		object->set_new_item_key(Variant());
```

Comment:
I'm not sure about the reasoon behind this change. I fully agree that it is strange from the start but it is not linked in anyway to typed dictionarry and there is no consensus on how to treat it so it should be removed from this PR.

Reply:
I suppose this could be its own PR, but it's baffling that this exists at all. You can already add null as a key in scripts & with an empty object, it's specifically a non-object null that the *editor* can't add

Reply:
I know I was too starting to become mad because of it but I feel this should have its own PR to discuss wether we want to remove it entirely change it etc. Your input would really be appreciated here : #88797 to expose what you think should be the solution and why you think this can't create errors contrary to what states the commentary.

Reply:
Migrated relevant code to #89117